src/client/chat.js

```diff
@@ -372,17 +412,27 @@ module.exports = function (client, options) {
       if (mcData.supportFeature('useChatSessions')) { // 1.19.3+
         const { acknowledged, acknowledgements } = getAcknowledgements()
         const canSign = client.profileKeys && client._session
-        client.write((mcData.supportFeature('seperateSignedChatCommandPacket') && canSign) ? 'chat_command_signed' : 'chat_command', {
+
+        // Prepare the packet parameters
+        const params = {
           command,
           timestamp: options.timestamp,
           salt: options.salt,
           argumentSignatures: canSign ? signaturesForCommand(command, options.timestamp, options.salt, options.preview, acknowledgements) : [],
           messageCount: client._lastSeenMessages.pending,
           acknowledged
-        })
+        }
+
+        // Add checksum for 1.21.5+
+        if (mcData.supportFeature('chatChecksum')) {
+          params.checksum = options.checksum !== undefined ? options.checksum : calculateChecksum(client._lastSeenMessages)
```

Comment:
nit: if both the if branches rely on same code you can avoid the duplicate code here by putting it up top